src/traces/image/plot.js

```diff
@@ -13,14 +13,24 @@ var Lib = require('../../lib');
 var xmlnsNamespaces = require('../../constants/xmlns_namespaces');
 var constants = require('./constants');
 
+function compatibleAxis(ax) {
+    return ax.type === 'linear' &&
+        // y axis must be reversed but x axis mustn't be
+        ((ax.range[1] > ax.range[0]) === (ax._id.charAt(0) === 'x'));
+}
+
 module.exports = function plot(gd, plotinfo, cdimage, imageLayer) {
     var xa = plotinfo.xaxis;
     var ya = plotinfo.yaxis;
 
+    var supportsPixelatedImage = !Lib.isIOS() && !Lib.isSafari() && !gd._context._exportedPlot;
```

Comment:
To reduce calls to check browser, I think you could 
```js
var isApple = !(Lib.isIOS() || Lib.isSafari());
```
outside the
```js
module.exports = function plot (...
```
and then here simply
```js
var supportsPixelatedImage = !(isApple || gd._context._exportedPlot);
```

Reply:
As suggested, I moved the browser check to the outside scope in fe1d2083c4ae61d7c9ee85561fb4654f4cde4643

Reply:
The downside is that now the test `does not render pixelated image when browser is not compatible` which relied on mocking the `navigator.userAgent` does not work anymore.